docs/make.jl

```diff
@@ -163,7 +163,8 @@ makedocs(
                  "Style guide" => "styleguide.md",
                  "Testing" => "testing.md",
                  "Performance" => "performance.md",
-                 "Parallelization" => "parallelization.md"
+                 "Parallelization" => "parallelization.md",
+                 "Interfacing with the p4est backend" => "p4est_interfacing.md"
```

Comment:
Does this really need to be a new section? I am wondering, since we already have a `P4estMesh` section that is focused on technical details: https://trixi-framework.github.io/Trixi.jl/stable/meshes/p4est_mesh/

Maybe this could just be the second large subsection there?

Reply:
So the reason why I put it down there was that in this new docu I heavily refer to undocumented/internal functions - I felt that this might be better placed in the "Advanced Topics & Developers" section. But I am fine with both.

Reply:
Hm. I agree, the current separation in "basic" and "advanced" is maybe not optimal, or at least not self-explanatory 😅 To me, "advanced topics" was just a catchall for anything that is not directly related to a concrete programming entity in Trixi.jl...

Reply:
I suggest to move it to the existing p4est docs for now, but we should keep an eye out for a new structure